### PR TITLE
Updates to make report upload work with latest orider portal changes

### PR DESCRIPTION
**Pull request by aanil**
Just wait a bit until I make black and prettier work on the new laptop.

### CODE REVIEW

File: daily_read/__main__.py [hunk 1]
```diff
@@ -64,22 +64,21 @@ def generate_all():
...
+    for orderer in orderer_with_modified_projects:
+        if orderer:
+            op.get_orders(orderer=orderer)
     # TODO - not sure how to proceed, fetching ALL projects in order portal
     # and then generate reports seems quite slow and wasteful, but I haven't
     # tried so it might work. Maybe a pull model where we only generate reports
     # for projects that we know have been updated can be a better alternative?
-    all_sthlm_orders = op.process_orders(use_node="Stockholm")
+    modified_orders = op.process_orders()
     # TODO: Should clean up...
```
Comment (alneberg) [inline daily_read/__main__.py]: Could you perhaps remove the comments that are no longer valid? Also goes for the TODO above, but that one might still be valid?

Comment (aanil) [reply daily_read/__main__.py]: I've removed the ones I think are done. The others are still valid.

File: daily_read/order_portal.py [hunk 2]
```diff
@@ -42,69 +44,69 @@ def get_orders(self, node=None, status=None, orderer=None, recent=True):
...
-                        }
+            if (
+                order["status"] == "closed"
+                and datetime.datetime.strptime(order["history"]["closed"], "%Y-%m-%d").date() >= older_than_cutoff
+            ):
+                continue
+            if order["identifier"] in self.projects_data.data.keys():
+                proj_info = self.projects_data.data[order["identifier"]]
+                if order["reports"]:
+                    proj_prog_rep = next(item for item in order["reports"] if item["name"] == "Project Progress")
```
Comment (alneberg) [inline daily_read/order_portal.py:79]: Not sure I follow the logic with the `next`. It returns one item from the list of project progress reports? Yeah, that might make sense. Perhaps add a comment?

File: daily_read/order_portal.py [hunk 3]
```diff
@@ -14,7 +15,7 @@
 class OrderPortal(object):
     """Class to handle NGI order portal interaction"""
 
-    def __init__(self, config_values):
+    def __init__(self, config_values, projects_data):
         self.config_values = config_values
         base_url = config_values.ORDER_PORTAL_URL
```
Comment (alneberg) [inline daily_read/order_portal.py:20]: I had an issue with urljoin removing the `/orders` part of my url `https://ngisweden.scilifelab.se/orders`. I had to add the trailing `/` for it to be kept. We should probably check that the base_url has a trailing `/`...

Comment (alneberg) [reply daily_read/order_portal.py:20]: I made an issue for later: https://github.com/NationalGenomicsInfrastructure/DailyRead/issues/17

File: daily_read/order_portal.py (hunk 2 above)
Comment (alneberg) [reply daily_read/order_portal.py:79]: I think this also will fail if there are no "Project Progress" report but there are other reports (rare occassion but still possible)

Comment (alneberg) [reply daily_read/order_portal.py:79]: Issue submitted:
https://github.com/NationalGenomicsInfrastructure/DailyRead/issues/18

**Review by alneberg — APPROVED** [review]
Great thank you! Converted comments to issues so we can deal with them later.

**Pull request merged**